scipy/stats/_wilcoxon.py

```diff
@@ -0,0 +1,261 @@
+    if has_zeros and method == "exact":
+        warnings.warn(message, stacklevel=2)
+
+    message = ("The sample size is small (fewer than 10 nonzero elements);"
+               "so `method='approx'` may not produce accurate p-values.")
+    count = d.shape[-1] - n_zero
+    too_small = np.any(count < 10)
+    if too_small and method == "approx":
+        warnings.warn(message, stacklevel=2)
+
+    if method == "auto":
+        if d.shape[-1] <= 50 and not has_zeros:
+            method = "exact"
+        elif not too_small:
+            method = "approx"
+        else:
+            method = stats.PermutationMethod()
+
+    return d, zero_method, correction, alternative, method, axis
+
+
+def _wilcoxon_statistic(d, zero_method='wilcox'):
+
+    i_zeros = (d == 0)
+
+    if zero_method == 'wilcox':
+        # Wilcoxon's method for treating zeros was to remove them from
+        # the calculation. We do this by replacing 0s with NaNs, which
+        # are ignored anyway.
+        d[i_zeros] = np.nan
+
+    i_nan = np.isnan(d)
+    n_nan = np.sum(i_nan, axis=-1)
+    count = d.shape[-1] - n_nan
+
+    r, t = rankdata(abs(d))
+    r_plus = np.sum((d > 0) * r, axis=-1)
+    r_minus = np.sum((d < 0) * r, axis=-1)
+
+    if zero_method == "zsplit":
+        # The "zero-split" method for treating zeros is to add half their contribution
+        # to r_plus and half to r_minus.
+        # See gh-2263 for the origin of this method.
+        r_zero_2 = np.sum(i_zeros * r, axis=-1) / 2
+        r_plus += r_zero_2
+        r_minus += r_zero_2
+
+    mn = count * (count + 1.) * 0.25
+    se = count * (count + 1.) * (2. * count + 1.)
+
+    if zero_method == "pratt":
+        # Pratt's method for treating zeros was just to modify the z-statistic.
+
+        # normal approximation needs to be adjusted, see Cureton (1967)
+        n_zero = i_zeros.sum(axis=-1)
+        mn -= n_zero * (n_zero + 1.) * 0.25
+        se -= n_zero * (n_zero + 1.) * (2. * n_zero + 1.)
+
+        # zeros are not to be included in tie-correction.
+        # any tie counts corresponding with zeros are in the 0th column
+        t[i_zeros.any(axis=-1), 0] = 0
+
+    tie_correct = (t**3 - t).sum(axis=-1)
+    se -= tie_correct/2
+    se = np.sqrt(se / 24)
+
+    # need to add continuity correction
+    z = (r_plus - mn) / se
+
+    return r_plus, r_minus, se, z, count
+
+
+def _correction_sign(z, alternative):
+    if alternative == 'greater':
+        return 1
+    elif alternative == 'less':
+        return -1
+    else:
+        return np.sign(z)
+
+
+def _wilcoxon_nd(x, y=None, zero_method='wilcox', correction=True, alternative='two-sided',
+                 method='auto', axis=0):
+    # test wilcox/pratt with all zeros
+    # change: respects specified method
+    temp = _wilcoxon_iv(x, y, zero_method, correction, alternative, method, axis)
+    d, zero_method, correction, alternative, method, axis = temp
+
+    r_plus, r_minus, se, z, count = _wilcoxon_statistic(d, zero_method)
+
```

Comment:
Should resolve `method='auto'` here. The statistic calculation doesn't depend on it, but the decision should be based on whether there are ties and/or zeros.